src/main/java/seedu/address/model/consultation/ConsultationList.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.model.consultation;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.consultation.exceptions.ConsultationNotFoundException;
+import seedu.address.model.consultation.exceptions.DuplicateConsultationException;
+
+/**
+ * A list of Consultation.
+ */
+public class ConsultationList implements Iterable<Consultation> {
+    private ObservableList<Consultation> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Consultation> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    public void setConsultationList(List<Consultation> consultationList) {
+        requireAllNonNull(consultationList);
+        internalList.setAll(consultationList);
+    }
+
+    /**
+     * Returns true if the list contains an equivalent consultation as the given argument.
+     */
+    public boolean contains(Consultation toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameConsultation);
+    }
+
+    /**
+     * Adds a consultation to the list.
+     */
+    public void addConsultation(Consultation toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateConsultationException();
+        }
+        internalList.add(toAdd);
+    }
+
+    //    /**
+    //     * Replaces the consultation {@code target} in the list with {@code editedConsultation}.
+    //     * {@code target} must exist in the list.
+    //     * The consultation {@code editedConsultation} must not be the same as another existing
+    //     * consultation in the list.
+    //     */
+    //    public void setConsult(Consultation target, Consultation editedConsultation) {
```

Comment:
not sure if this was left commented out intentionally

Reply:
Removed deadcode for next milestone